File: sprint2/test_lecture.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream fichier("fichier.txt");
    std::string ligne;
    int lignes_compte = 0;

    if (fichier.is_open()) {
        while (getline(fichier, ligne)) {
            lignes_compte++;
        }
        fichier.close();
        std::cout << "Nombre de lignes : " << lignes_compte << "\n";
    } else {
        std::cout << "Impossible d'ouvrir le fichier\n";
    }

    return 0;
}

File: sprint2/TestLecture.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class TestLecture {
    public static void main(String[] args) {
        int lignesCompte = 0;
        
        try (BufferedReader br = new BufferedReader(new FileReader("fichier.txt"))) {
            String ligne;
            while ((ligne = br.readLine()) != null) {
                lignesCompte++;
            }
            System.out.println("Nombre de lignes : " + lignesCompte);
        } catch (IOException e) {
            System.out.println("Erreur de lecture : " + e.getMessage());
        }
    }
}